File: Searching/BinarySearch.java
package Searching;
import java.util.Arrays;

public class BinarySearch {
    public static int binarySearch(int[] arr, int key) {
        int left = 0, right = arr.length - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;

            // Check if key is present at mid
            if (arr[mid] == key)
                return mid;

            // If key greater, ignore left half
            if (arr[mid] < key)
                left = mid + 1;

            // If key is smaller, ignore right half
            else
                right = mid - 1;
        }
        return -1; // Element not found
    }

    public static void main(String[] args) {
        int[] arr = {2, 3, 4, 10, 40};
        Arrays.sort(arr);  // Ensure array is sorted before using binary search
        int key = 10;
        int result = binarySearch(arr, key);
        if (result == -1) {
            System.out.println("Element not found");
        } else {
            System.out.println("Element found at index: " + result);
        }
    }
}

File: DSA/Python/Binary Search.py
def binary_search(arr, key):
    low = 0
    high = len(arr) - 1
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == key:
            return mid  # Key found at mid index
        elif arr[mid] < key:
            low = mid + 1  # Search in the right half
        else:
            high = mid - 1  # Search in the left half
    return -1  # Return -1 if key is not found

arr = [11, 15, 23, 45, 70]
key = 15
result = binary_search(arr, key)

if result != -1:
    print(f"Element found at index {result}")
else:
    print("Element not found")